components/kyma-environment-broker/internal/lms/client.go

```diff
@@ -0,0 +1,398 @@
+package lms
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	random "math/rand"
+	"net/http"
+
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/iosafety"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type Client interface {
+	CreateTenant(input CreateTenantInput) (o CreateTenantOutput, err error)
+	GetTenantStatus(tenantID string) (status TenantStatus, err error)
+	GetTenantInfo(tenantID string) (status TenantInfo, err error)
+
+	GetCACertificate(tenantID string) (cert string, found bool, err error)
+	GetSignedCertificate(tenantID string, certID string) (cert string, found bool, err error)
+	RequestCertificate(tenantID string, subject pkix.Name) (string, []byte, error)
+}
+
+type Environment string
+
+type ClusterType string
+
+type client struct {
+	url string
+
+	// tenant predefined values
+	clusterType ClusterType
+	environment Environment
+
+	token      string
+	samlTenant string
+
+	log logrus.FieldLogger
+}
+
+const (
+	EnvironmentDev  Environment = "dev"
+	EnvironmentProd Environment = "prod"
+
+	ClusterTypeHA         ClusterType = "ha"
```

Comment:
unused, will it be implemented in the future?

Reply:
I'll add validation to use it